homeassistant/components/arcam_fmj/__init__.py

```diff
@@ -35,44 +23,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-
-def _optional_zone(value):
-    if value:
-        return ZONE_SCHEMA(value)
-    return ZONE_SCHEMA({})
-
-
-def _zone_name_validator(config):
-    for zone, zone_config in config[CONF_ZONE].items():
-        if CONF_NAME not in zone_config:
-            zone_config[
-                CONF_NAME
-            ] = f"{DEFAULT_NAME} ({config[CONF_HOST]}:{config[CONF_PORT]}) - {zone}"
-    return config
-
-
-ZONE_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_NAME): cv.string,
-        vol.Optional(SERVICE_TURN_ON): cv.SERVICE_SCHEMA,
-    }
-)
-
-DEVICE_SCHEMA = vol.Schema(
-    vol.All(
-        {
-            vol.Required(CONF_HOST): cv.string,
-            vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.positive_int,
-            vol.Optional(CONF_ZONE, default={1: _optional_zone(None)}): {
-                vol.In([1, 2]): _optional_zone
-            },
-            vol.Optional(
-                CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-            ): cv.positive_int,
-        },
-        _zone_name_validator,
-    )
-)
+CONFIG_SCHEMA = cv.deprecated(DOMAIN, invalidation_version="0.115")
```

Comment:
Isn't this invalid now ?

Reply:
Yes, but don't want upgrades to be blocked. One option would be to remove all validation, but then the config would be remain stale in people's config since no warning would be emitted.

Reply:
Seems fine as-is